src/CMakeLists.txt

```diff
@@ -301,9 +301,11 @@ endif()
 # However, Parthenon heavily relies on it and there is no harm in compiling Kokkos
 # without and Parthenon with (via Max Katz on the Kokkos Slack channel).
 # Therefore, we don't use the Kokkos_ENABLE_CUDA_CONSTEXPR option add the flag manually.
-# Also, not checking for NVIDIA as nvcc_wrapper is identified as GNU so we just make sure
-# the flag is not added when compiling with Clang for Cuda.
-if (Kokkos_ENABLE_CUDA AND NOT CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+# Also, not checking for NVIDIA as nvcc_wrapper is identified as GNU so we just
+# make sure the flag is not added when compiling with Clang for Cuda. The flag
+# is also not necessary (and not recognized) when using nvc++ to compile Cuda
+# code
+if (Kokkos_ENABLE_CUDA AND NOT CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND NOT CMAKE_CXX_COMPILER_ID STREQUAL "NVHPC")
```

Comment:
This makes me a little nervous, as it may cause problems with downstream codes.

Reply:
Maybe this condition could be changed to check whether `CMAKE_CUDA_COMPILER_ID` is not equal to `NVHPC` instead?

Reply:
Sure---that seems like it might be safer. My main worry here is will the jury-rigged insanity that I (and KHARMA) are currently using to build with nvhpc fail?

Reply:
Hmm, I don't know. I'm not sure how that works at all without removing this flag, unless you mean you're using nvc++ as the host compiler and nvcc as the device compiler? In that case, I think it _should_ still report `NVIDIA` as the CUDA compiler, rather than `NVHPC`.

Reply:
I don't think so. I feel like we've discussed this before. The option `--expt-relaxed-constexpr` is specific to the `nvcc` compiler, it's not an option for `nvc++`. I'm double checking that  `nvc++`  can compile CUDA code that needs `constexpr`.

Does `CMAKE_CUDA_COMPILER_ID` get set to `NVHPC` or `CLANG`? The current commit also only changes the behavior of `nvc++` builds, I don't think we want to change the behavior of `clang` builds.

Reply:
If you use clang as the device compiler, it doesn't accept this option either (it doesn't need it). So IMO it might be good to keep that.

Reply:
I'm not sure if you can force `nvc++` to use `nvcc` as the CUDA compiler, `nvc++` already includes the CUDA toolchain.

I also verified that `nvc++` compiles and runs code that needs `nvcc --expt-relaxed-constexpr`.

Reply:
By default, at least on non-Cray systems, it uses `nvcc` as the CUDA compiler. It only uses nvc++ for device code when `nvc++ -cuda` (or `nvc++ -stdpar` or `nvc++ -openacc`) is used.

Reply:
I don't remember the full context for this discussion - is this issue resolved?